test/__snapshots__/index.js.snap

```diff
@@ -67,6 +67,54 @@ const Test = () => <div className={\\"jsx-2529315885\\"}>
   </div>;"
 `;
 
+exports[`works with array children 1`] = `
+"import _JSXStyle from \\"styled-jsx/style\\";
+/* eslint-disable no-unused-vars */
+
+const Test1 = () => [<div className={\\"jsx-336794318\\"}>Foo</div>, <span className={\\"jsx-336794318\\"}>Bar</span>, <_JSXStyle styleId={\\"1486255031\\"} css={\\"div.jsx-336794318{color:red}\\"} />];
+
+const Test2 = () => [<p className={\\"jsx-136928739\\"}>Hello World</p>, <_JSXStyle styleId={\\"2639504339\\"} css={\\"p{color:white}\\"} />];
+
+const Test3 = () => [<div className={\\"jsx-3360849707\\"}>
+    <p className={\\"jsx-3360849707\\"}>Hello World</p>
+    <_JSXStyle styleId={\\"4193872369\\"} css={\\"p.jsx-3360849707{color:blue}\\"} />
+  </div>, <div>Hello World</div>];
+
+const Test4 = () => [<div className={\\"jsx-612908104\\"}>
```

Comment:
Alright @giuseppeg so to start off I wanted to get a new baseline of where we currently stand.

After upgrading the dependencies and the snapshots, it looks like Test4 is the only place where it seems to still be stumbling. Specifically, the issue is that there is a decision between what style to scope into the selector if there are both a top-level style element and inner style elements.

I'll ping you on slack to discuss what you want to do about cases like this.